src/ioctls/device.rs

```diff
@@ -207,5 +285,24 @@ mod tests {
 
         assert!(device_fd.has_device_attr(&dist_attr).is_ok());
         assert!(device_fd.set_device_attr(&dist_attr).is_ok());
+
+        // Test `get_device_attr`. Here we try to extract the maximum supported number of IRQs.
+        // This value should be saved in the address provided to the ioctl.
+        let mut data: u32 = 0;
+
+        let mut gic_attr = kvm_bindings::kvm_device_attr::default();
+        gic_attr.group = KVM_DEV_ARM_VGIC_GRP_NR_IRQS;
+        gic_attr.addr = data as u64;
+
+        // Without properly providing the address to where the
+        // value will be stored, the ioctl fails with EFAULT.
+        let res = device_fd.get_device_attr(&mut gic_attr);
+        assert!(res.is_err() && errno::Error::last().errno() == libc::EFAULT);
```

Comment:
`res` is already `errno::Error::last`. No need to call that again. I believe that it is actually incorrect to call it again. If you add any ioctl between line 299 and line 300 than errno will be the error of that last call and not the result from `get_device_attr`. If this happens in another part of the kvm, I can open an issue with that.

Reply:
Thanks @andreeaflorescu for explaining! Then I think I should remove the assert of error code here and only leave the assertion of calling this ioctl.

Reply:
Sure, thanks for your PR!